api/cpp/include/slint.h

```diff
@@ -589,6 +588,14 @@ class Model
         model_row_data_dirty_property.get();
     }
 
+    /// \private
```

Comment:
Oh, I see this is marked as private API. I guess that's not intentional?

Reply:
Since track_row_data_changes is currently private, i thought we didn't want to make this public for now.

Reply:
I'm not entirely sure I understand :-)

This patch:
 * Adds public `row_data_tracked` function to the Rust `Model` trait, as per request by a user.
 * Adds C++ equivalent to `Model` C++ class, but it's private and internal.

Why not make the new data getter function public in C++ as well?

Reply:
`track_row_data_changes` had an empty documentation but it is still public API in Rust.  But the equivalent in C++ is marked as internal.

`row_data_tracked` is a convenience for track_row_data_changes. Since the later is public in rust, i made the formar public. In C++, since track_row_data_changes is private, i made row_data_tracked as private as well.

Reply:
You're right, your change is perfectly consistent :-). I'm fine with it remaining private (go ahead and merge), just thinking that we'll make it public anyway once a C++ user files #1177 :-)